### PR TITLE
Human readable dates and multiple file path support in the collate command

### PR DESCRIPTION
**Pull request by rhyshort**
- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)

## What

- Make `--start` and `--end` options in collate accept humna readable dates, ie 2022-09-29 instead of 20220929
- Enable collation of mulitple files with a single collate command by using `,` as a path seperator.

## Why

The current `--start` and `--end` formats are hard for humans to read, which makes collating data to provide to auditors difficult, as conversion between date formats needs to be done by hand, and the number grouping makes it hard to ensure there have not been any typos etc

Being able to pull multiple files at once makes it easier to pull multiple files over the same date range when an auditor is requesting multiple files.

## How

- Replace all references to `YYYYMMDD` dates to `YYYY-MM-DD` dates
- Split the file path string and iterate over the result calling collate functions multiple times.

## Test

- Updated tests for the new date format
- Add a new test to pull multiple files.

### CODE REVIEW

File: harvest/cli.py
Comment (drsm79) [inline harvest/cli.py]: I think better would be to use an `append` action, so the list is explicit

**Review by drsm79 — CHANGES_REQUESTED** [review]

File: harvest/cli.py
Comment (drsm79) [reply harvest/cli.py]: https://docs.python.org/3/library/argparse.html

Comment (rhyshort) [reply harvest/cli.py]: `append` action wasn't doing quite the right thing with postional args, so option to use `nargs='+'`, which has the desired behaviour. Example help text: 
```
harvest collate --help
usage: harvest collate [-h] [--repo-path ~/path/git-repo] [--creds ~/path/creds] [--end YYYY-MM-DD] [--start YYYY-MM-DD] [--version] repo filepath [filepath ...]

positional arguments:
  repo                  the URL to the repository containing files to be processed by harvest, as an example: https://github.com/my-org/my-repo or: local - if working exclusively with a local git repo
  filepath              the relative path to a file in a git repository that you wish to retrieve.

optional arguments:
  -h, --help            show this help message and exit
  --repo-path ~/path/git-repo
                        the operating system location of a local git repository - if not provided, repo path is assumed to be $TMPDIR/harvest
  --creds ~/path/creds  the path to credentials file - defaults to ~/.credentials
  --end YYYY-MM-DD      the end of date range for the file you wish to retrieve - defaults to the current date
  --start YYYY-MM-DD    the start of date range for the file you wish to retrieve - defaults to same value as the end of date range
  --version             the harvest version```

File: harvest/cli.py
Comment (drsm79) [inline harvest/cli.py]: Sure I mentioned that this is an unnecessary breaking change. Suggest we have a function that takes a date in either YYYYMMDD or YYYY-MM-DD format and returns the datetime object for it. That will preserve current behaviour for anyone who's got a script using the tool and support both ISO date formats.

Comment (rhyshort) [reply harvest/cli.py]: took me some time to get around to it but done now

Comment (drsm79) [reply harvest/cli.py]: Nice one

**Review by drsm79 — DISMISSED** [review]

**Review by drsm79 — DISMISSED** [review]

**Comment by rhyshort** [comment]
hmm the github action looks like is failing because of the python version is too old for the version of flake8.

**Comment by cletomartin** [comment]
@rhyshort I'm changing all the linting side of the project here #28

**Comment by rhyshort** [comment]
@cletomartin nice, I'll rebase this on top of #28 once merged.

**Comment by rhyshort** [comment]
Rebased on top of the changes from #28

**Review by cletomartin — COMMENTED** [review]
Thanks!

Please update the version https://github.com/ComplianceAsCode/auditree-harvest/blob/main/harvest/__init__.py#L16 and the CHANGES.md file too.

**Comment by rhyshort** [comment]
Done

**Review by cletomartin — APPROVED** [review]

**Comment by rhyshort** [comment]
Thanks @cletomartin, I don't have write access so can you go ahead and merge please?

**Comment by drsm79** [comment]
Merged - thanks Rhys!

**Pull request merged**
